### PR TITLE
SAK-50371 Search upgrade to opensearch 2 and lucene 9

### PR DESCRIPTION
**Pull request by ern**
https://sakaiproject.atlassian.net/browse/SAK-50371

### CODE REVIEW

File: microsoft-integration/impl/pom.xml [hunk 1]
```diff
@@ -51,12 +51,14 @@
 
 		<!-- Microsoft dependencies -->
 		<dependency>
-			<groupId>com.azure</groupId>
-			<artifactId>azure-identity</artifactId>
```
Comment (ern) [inline microsoft-integration/impl/pom.xml:55]: When I was reviewing libs I noticed that azure-identity sdk may not actually be needed as it only appears to be using msal4j.  Would need to test and make sure that this is accurate.

Comment (bgarciaentornos) [reply microsoft-integration/impl/pom.xml:55]: We'll check it and come back to you once we ensure it keeps working like that, thanks!

Comment (bgarciaentornos) [reply microsoft-integration/impl/pom.xml:55]: We've done some testing and everything seems to be working as expected with these changes, thanks!

File: search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/SiteElasticSearchIndexBuilder.java [hunk 2]
```diff
@@ -380,7 +375,7 @@ protected void deleteAllDocumentForSite(String siteId) {
         // try {
         //     client.deleteByQuery(request, RequestOptions.DEFAULT);
         // } catch (IOException ioe) {
-        //     getLog().warn("Could not delete all documents in index {} for site {}, {}", indexName, siteId, ioe.toString());
+        //     log.warn("Could not delete all documents in index {} for site {}, {}", indexName, siteId, ioe.toString());
         // }
```
Comment (bgarciaentornos) [inline search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/SiteElasticSearchIndexBuilder.java:379]: Maybe we can remove all this?

File: search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/BaseElasticSearchIndexBuilder.java [hunk 3]
```diff
@@ -1050,14 +1036,14 @@ else if (propertyValue instanceof Collection)
...
+                    log.warn("Couldn't find what the value for '{}' was. It has been ignored. {}", propertyName, propertyName.getClass());
                 values = Collections.emptyList();
             }
 
             //If this property was already present there (this shouldn't happen, but if it does everything must be stored
             if (properties.containsKey(propertyName)) {
-                getLog().warn("Two properties had a really similar name and were merged. This shouldn't happen! " + propertyName);
-                getLog().debug("Merged values '" + properties.get(propertyName) + "' with '" + values);
+                log.warn("Two properties had a really similar name and were merged. This shouldn't happen! " + propertyName);
+                log.debug("Merged values '" + properties.get(propertyName) + "' with '" + values);
```
Comment (bgarciaentornos) [inline search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/BaseElasticSearchIndexBuilder.java:1046]: Use new logging format

**Review by bgarciaentornos — APPROVED** [review]
Looks great, just a couple small things. Approving it anyway. Thanks!

File: search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/BaseElasticSearchIndexBuilder.java (hunk 3 above)
Comment (ern) [reply search/elasticsearch/util/src/java/org/sakaiproject/search/elasticsearch/BaseElasticSearchIndexBuilder.java:1046]: thanks Bernie for catching that, I missed that one!

File: microsoft-integration/impl/pom.xml (hunk 1 above)
Comment (ern) [reply microsoft-integration/impl/pom.xml:55]: awesome thanks for the testing

**Pull request merged**